pumpkin-cli/src/flatzinc/compiler/post_constraints.rs

```diff
@@ -256,7 +256,7 @@ fn compile_array_int_maximum(
     let rhs = context.resolve_integer_variable(&exprs[0])?;
     let array = context.resolve_integer_variable_array(&exprs[1])?;
 
-    Ok(constraints::maximum(array.as_ref(), rhs)
+    Ok(constraints::maximum(array.as_ref().to_owned(), rhs)
```

Comment:
Why this `to_owned`?

Reply:
As discussed, this has to do with the lifetimes passing through constraint poster, and rust not being able to statically guarantee that the array will be alive when the constraint is posted.